kolibri/plugins/hooks.py

```diff
@@ -315,6 +337,37 @@ def add_hook_to_class_registry(cls, hook):
             "{} added to registry for defined hook: {}".format(hook.unique_id, cls)
         )
 
+    def remove_hook_from_registries(cls):
+        """
+        Remove a concrete hook class from all relevant abstract hook registries.
+        """
+        if not cls.abstract and cls._registered:
+            hook = cls()
+            for parent in cls.__mro__:
+                if (
+                    isabstract(parent)
+                    and issubclass(parent, KolibriHook)
+                    and parent is not KolibriHook
+                    and hasattr(parent, "_registered_hooks")
+                ):
```

Comment:
Not a blocker: some of this logic seems like it's getting a bit repetitive. There are subtle differences, but bulk of it seems like it could be consolidated. Every time I look at this area of the codebase, it takes me a while to carefully understand all of the conditions. Some consolidation can tuck that a way into a well commented chunk of code, so each usage can focus on its goal and what's different